### PR TITLE
 Add leaflet-amplify

### PR DESCRIPTION
**Pull request by dayjournal**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

This example demonstrates how to use Leaflet with Amazon Location Service and AWS Amplify.
https://dev.to/aws-heroes/building-a-map-application-with-amazon-location-service-leaflet-and-aws-amplify-5gbe

### CODE REVIEW

File: leaflet-amplify/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+![](./img/README02.png)
+
+This example demonstrates how to use Leaflet with Amazon Location Service and AWS Amplify.
+
+## Execution environment
+
+- node v18.x
+- npm v8.x
+- Amplify CLI v9.2.x
```
Comment (dreamorosi) [inline leaflet-amplify/README.md:13]: You have used version v10.x to create the Amplify resources according to the config file in the `amplify` folder.

I'm not 100% sure if they're compatible, so either:
- Change the min version in the readme to match the one you have used.
- OR test the sample with Amplify CLI 9.2.x and confirm it works.

**Review by dreamorosi — CHANGES_REQUESTED** [review]
Hi @dayjournal thank you for the PR, I have left a couple of comments.

File: leaflet-amplify/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  "version": "1.9.2",
+  "description": "",
+  "scripts": {
+    "dev": "vite",
+    "build": "tsc && vite build",
+    "preview": "vite preview"
+  },
+  "keywords": [],
+  "author": "Yasunori Kirimoto",
+  "license": "ISC",
```
[the remark this answers is not in the document]
Comment (dayjournal) [reply leaflet-amplify/package.json]: Fixed.

File: leaflet-amplify/README.md (hunk 1 above)
Comment (dayjournal) [reply leaflet-amplify/README.md]: Fixed.

**Comment by dayjournal** [comment]
Hi, @dreamorosi. Thanks for checking. I have modified my PR.

**Review by dreamorosi — CHANGES_REQUESTED** [review]
Thank you for making the changes, appreciate it.

I was finally able to test the code on my machine, and it works.

The only other change we should do is update these two dependencies like this:
```
 "maplibre-gl": "^2.4.0",
"maplibre-gl-js-amplify": "^2.0.4"
```

Currently running `npm audit` results in this:
```sh
npm audit                                                                                                                                                                                                                                                                                                                              <aws:aamorosi-Admin>
# npm audit report

minimist  <=1.2.5
Severity: critical
Prototype Pollution in minimist - https://github.com/advisories/GHSA-xvch-5gv4-984h
Prototype Pollution in minimist - https://github.com/advisories/GHSA-vh95-rmgr-6w4m
Prototype Pollution in minimist - https://github.com/advisories/GHSA-vh95-rmgr-6w4m
fix available via `npm audit fix --force`
Will install maplibre-gl-js-amplify@2.0.4, which is a breaking change
node_modules/@mapbox/geojsonhint/node_modules/minimist
node_modules/geojson-rewind/node_modules/minimist
node_modules/mapbox-gl/node_modules/minimist
node_modules/maplibre-gl-draw-circle/node_modules/minimist
node_modules/sharkdown/node_modules/minimist
  @mapbox/geojsonhint  <=3.0.0
  Depends on vulnerable versions of minimist
  node_modules/@mapbox/geojsonhint
    @mapbox/mapbox-gl-draw  0.16.1 - 1.2.2
    Depends on vulnerable versions of @mapbox/geojsonhint
    node_modules/maplibre-gl-draw-circle/node_modules/@mapbox/mapbox-gl-draw
      maplibre-gl-draw-circle  <=0.0.6
      Depends on vulnerable versions of @mapbox/mapbox-gl-draw
      node_modules/maplibre-gl-draw-circle
        maplibre-gl-js-amplify  1.2.2-geo - 1.2.2-geo.1 || 1.2.6-geo - 1.2.9-geo || 1.4.0 - 1.4.2-unstable.2 || 1.5.1-geo.1 - 2.0.2-geo.1
        Depends on vulnerable versions of maplibre-gl-draw-circle
        node_modules/maplibre-gl-js-amplify
  geojson-flatten  0.2.2 - 1.0.3
  Depends on vulnerable versions of minimist
  node_modules/maplibre-gl-draw-circle/node_modules/geojson-flatten
  geojson-rewind  >=0.3.0
  Depends on vulnerable versions of minimist
  Depends on vulnerable versions of sharkdown
  node_modules/geojson-rewind
    mapbox-gl  0.39.1 - 1.9.1
    Depends on vulnerable versions of geojson-rewind
    Depends on vulnerable versions of minimist
    node_modules/mapbox-gl
  sharkdown  *
  Depends on vulnerable versions of minimist
  node_modules/sharkdown

underscore  1.3.2 - 1.12.0
Severity: critical
Arbitrary Code Execution in underscore - https://github.com/advisories/GHSA-cf4h-3jhx-xvhq
fix available via `npm audit fix`
node_modules/underscore
  nomnom  >=1.6.0
  Depends on vulnerable versions of underscore
  node_modules/nomnom

11 vulnerabilities (8 moderate, 3 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force
```

I tested after the update & it works + there's no vulnerability

**Comment by dayjournal** [comment]
Hi, @dreamorosi 
In my environment, "@maplibre/maplibre-gl-leaflet" is incompatible with maplibre-gl ver2, so I get an error.

It works fine with the following version.
```
"maplibre-gl":"^1.15.3",
"maplibre-gl-js-amplify":"^1.6.0"
```

Error Description
```
bash-3.2$ npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: leaflet-amplify@1.9.2
npm ERR! Found: maplibre-gl@2.4.0
npm ERR! node_modules/maplibre-gl
npm ERR!   maplibre-gl@"^2.4.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer maplibre-gl@"^1.14.0" from @maplibre/maplibre-gl-leaflet@0.0.17
npm ERR! node_modules/@maplibre/maplibre-gl-leaflet
npm ERR!   @maplibre/maplibre-gl-leaflet@"^0.0.17" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/dayjournal/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
```

**Comment by dreamorosi** [comment]
You need to delete the `package-lock.json` file & `node_modules` folder before trying to change the versions.

Give it a try and if it doesn't work I'll merge the PR and do the update afterwards.

**Comment by dayjournal** [comment]
@dreamorosi 
I tried, but it didn't work correctly...

**Comment by dreamorosi** [comment]
Fixed in the [latest commit](https://github.com/aws-samples/amazon-location-samples/commit/3c4f25239faa6b88878d68cc97f8d9eaa3221027) on `main`.

It worked by doing this:
- remove existing `node_modules` & `package-lock.json`
- remove `maplibre-gl-js-amplify` & `maplibre-gl-js` from `package.json`
- run `npm I`
- install these two dependencies afterwards

At this point the `package-lock.json` file is generated with all the dependencies.

---

Thank you for another cool contribution @dayjournal 🎉

**Pull request merged**